sympy/concrete/summations.py

```diff
@@ -233,11 +233,7 @@ def _eval_derivative(self, x):
         # don't want to differentiate wrt any free symbol in the upper or lower
         # limits
         # XXX remove this test for free_symbols when the default _eval_derivative is in
-        if x.is_Indexed:
-            from sympy import IndexedBase
-            if x.base not in self.atoms(IndexedBase):
-                return S.Zero
-        elif x not in self.free_symbols:
+        if isinstance(x, Symbol) and x not in self.free_symbols:
```

Comment:
Point 1 in your issue. Constant-ness check for instances of Symbol only. For the rest, let the derivative be dispatched.